catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchSource.java

```diff
@@ -179,6 +179,30 @@ public OpenSearchSource(
     this.encryptionService = encryptionService;
     this.openSearchParser = openSearchParser;
     this.openSearchFilterVisitor = openSearchFilterVisitor;
+    this.foreignMarkupBiConsumer = (elements, sourceResponse) -> {};
+  }
+
+  /**
+   * Creates an OpenSearch Site instance. Sets an initial default endpointUrl that can be
+   * overwritten using the setter methods.
+   */
+  public OpenSearchSource(
+      FilterAdapter filterAdapter,
+      OpenSearchParser openSearchParser,
+      OpenSearchFilterVisitor openSearchFilterVisitor,
+      EncryptionService encryptionService,
+      BiConsumer<List<Element>, SourceResponse> foreignMarkupBiConsumer) {
+    this.filterAdapter = filterAdapter;
+    this.encryptionService = encryptionService;
+    this.openSearchParser = openSearchParser;
+    this.openSearchFilterVisitor = openSearchFilterVisitor;
+    this.foreignMarkupBiConsumer = foreignMarkupBiConsumer;
+  }
+
+  /** Package-private so that tests may set the foreign markup consumer. */
+  void setForeignMarkupBiConsumer(
```

Comment:
✏️  Having this organized near the other getters/setters in the file would be nice

Reply:
done